### ISSUE TITLE
Interacting with Dumpert.

### ISSUE THREAD

**Issue by CytoDev** [jariz/Dumpert#35]
Commenting/replying/voting/etc.  
  
I'll be working on it in the newNavDrawer branch.  
  
The new drawer is done, bytheway.

**Comment by CytoDev** [comment]
### Voting on items and comments
Voting on comments has been implemented, but could be done better (like catching a server response and updating the number based on that). Voting on items seems to link me to [this page](https://www.zazzle.com/rlv/i_am_gay_t_tshirt-p2357526868521500423yuv_400.jpg) instead of something useful (Thanks ~~Obama~~ Joris!).  
  
See [#8cbf8fc](https://github.com/jariz/Dumpert/commit/8cbf8fc9a6cdb4d96528e6a169524196bea54fe1), [#ff458f2](https://github.com/jariz/Dumpert/commit/ff458f25efb12b0c5d183521cb3768cd34b1afd2), and [#e5b097c](https://github.com/jariz/Dumpert/commit/e5b097cb44050fa3715cb722c59805995862540e)  
  
### Replying to items and comments
Comments are sent to `http://app.steylloos.nl/mt-comments.fcgi`. Request cookies are `__cfduid`, `commenter_name`, and `tk_commenter`. No interesting headers are sent, and the comment is encoded as `application/x-www-form-urlencoded`.  
  
This means that replying is easily implemented in theory. However, like voting on items proves, Joris might have another nice t-shirt in store for us.  
  
### Question regarding replying to comments
@jariz @KeizerDev @gewoonrik and everyone I forgot to tag:  
How do we want replying on comments to work? There doesn't seem to be any consistency on Dumpert itself. Some users use @ references while others copy the entire comment and type their response above or below.  
  
I would suggest having both options available, but having users select their preferred way from the settings. Any inputs on your sides?

**Comment by KeizerDev** [comment]
Why not make an option for copying the whole comment on longpress and just simply past it in the comment field. Or just give an option replay and just auto-past everything in the comment field. I prefer the selecting thing rather than `@ mention`

**Comment by CytoDev** [comment]
@KeizerDev I opted to use the copy paste way of replying. Should we allow users to select their preference or just force them to do it like this? They can just delete it and type the `@ mention` if they really want to either way.

**Comment by CytoDev** [comment]
@jariz @KeizerDev @gewoonrik Created [voting branch](https://github.com/jariz/Dumpert/tree/voting). Better nate than lever.

**Comment by KeizerDev** [comment]
@CytoDev No, force them to do it this way. Like you said, they can just ignore our way and use the `@`

**Comment by CytoDev** [comment]
@KeizerDev Less work for us hahah :smile:

**Comment by KeizerDev** [comment]
@CytoDev That's also true :smiley:

**Comment by CytoDev** [comment]
# some additional info's:
## request headers when upvoting a comment
    Host: www.geenstijl.nl
    User-Agent: Mozilla/5.0 (X11; Debian_C0.1; Linux x86_64; rv:43.0) Gecko/20100101 Firefox/43.0
    Accept: */*
    Accept-Language: en-US,en;q=0.5
    Accept-Encoding: gzip, deflate
    DNT: 1
    Referer: http://www.dumpert.nl/mediabase/6711758/eac8f1aa/emma_belt_met_reaguurders.html
    Cookie: __cfduid=d193b1ef9b771f73cf30b7eba17cf1adf1452084513; cpc=10; wsLcuid=414461315055232424; commenter_name=terreurhaas; tk_commenter=VFn3KM+EhIGb4hiIDnL6suvnIvk%3D%3AvWOn9+e63UTBt%2Fs354psiuL04KE%3D
    Connection: keep-alive
    Pragma: no-cache
    Cache-Control: no-cache
## request headers when downvoting a post
    Host: www.dumpert.nl
    User-Agent: Mozilla/5.0 (X11; Debian_C0.1; Linux x86_64; rv:43.0) Gecko/20100101 Firefox/43.0
    Accept: application/json, text/javascript, */*; q=0.01
    Accept-Language: en-US,en;q=0.5
    Accept-Encoding: gzip, deflate
    DNT: 1
    X-Requested-With: XMLHttpRequest
    Referer: http://www.dumpert.nl/mediabase/6711758/eac8f1aa/emma_belt_met_reaguurders.html
    Cookie: __cfduid=da3d3184efad40ae43481ebc54b90ac1f1452084496; cpc=10; wsLcuid=414461222119551041; __utma=73838087.875357062.1452084497.1452852941.1452856884.17; __utmz=73838087.1452084497.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none); rsi_segs=G07609_10692|G07609_10697|G07609_10729|G07609_10738|G07609_10827|G07609_10831|G07609_10833|G07609_10892|G07609_10908|G07609_0; commenter_name=terreurhaas; tk_commenter=VFn3KM+EhIGb4hiIDnL6suvnIvk%3D%3AvWOn9+e63UTBt%2Fs354psiuL04KE%3D; __utmc=73838087; __utmb=73838087.12.9.1452856905585; __utmt=1
    Connection: keep-alive
    Pragma: no-cache
    Cache-Control: no-cache

I'm quite sure most of those cookies are unavailable when connecting to a post via the app. I tried fiddling with the request headers but had no luck.  
  
@jariz kan jij hier wat mee?

**Comment by jariz** [comment]
Voting removed for now (#54), will be fixed in seperate voting branch later.
@CytoDev Yes, zal er binnenkort even naar kijken, thanks.

**Comment by CytoDev** [comment]
[for quick reference](https://github.com/jariz/Dumpert/blob/voting/Dumpert/src/main/java/io/jari/dumpert/api/API.java#L475-L506)
